symforce/slam/imu_preintegration/preintegrated_imu_measurements.h

```diff
@@ -0,0 +1,96 @@
+/* ----------------------------------------------------------------------------
+ * SymForce - Copyright 2022, Skydio, Inc.
+ * This source code is under the Apache 2.0 license found in the LICENSE file.
+ * ---------------------------------------------------------------------------- */
+
+#pragma once
+
+#include <Eigen/Dense>
+
+#include <sym/rot3.h>
+#include <sym/util/epsilon.h>
+
+namespace sym {
+
+/**
+ * Struct of Preintegrated IMU Measurements (not include the covariance of change in orientation,
+ * velocity, and position).
+ */
+template <typename Scalar>
+struct PIMNoCovariance {
```

Comment:
I have another idea to throw in:

```cpp
struct PreintegratedMeasurements {
  sym::Rot3<Scalar> DR;
  Vector3 Dv;
  Vector3 Dp;
  Matrix3 DR_D_gyro_bias;
  Matrix3 Dv_D_gyro_bias;
  Matrix3 Dv_D_accel_bias;
  Matrix3 Dp_D_gyro_bias;
  Matrix3 Dp_D_accel_bias;
  Vector3 accel_bias;
  Vector3 gyro_bias;

  double integrated_dt;
};

class Preintegrator {
  private:
    PreintegratedMeasurements preintegrated_measurements_;
    Matrix99 covariance_;

  public:
    void AddMeasurement(stuff);
    const PreintegratedMeasurements& PreintegratedMeasurements() const;
    const Matrix99& Covariance() const;
};

class ImuFactor {
  private:
    PreintegratedMeasurements preintegrated_measurements_;
    Matrix99 sqrt_info_;

  public:
    ImuFactor(PreintegratedMeasurements preintegrated_measurements, Matrix99 covariance);
};
```

I'm generally not worried about the 81 doubles (I think it's both a smaller percentage of "memory per IMU factor" than this is counting, assuming the 57% from before is the size of the memory in or owned by the `ImuFactor` object itself, and is likely small compared to perception data these factors are typically used with).  I'm not opposed to trying to save it though since it isn't too much work.

Is there a particular place you're worried about things becoming very verbose?  The usages of `pim` I see in this file seem like they'd be fine as `preintegrated_measurements_`

Reply:
Ignoring unit tests, really the only usage would be in `imu_factor.cc`

Reply:
Anyway, I've changed it to be more like how you have it. One difference though is that I kept `AddMeasurment` named `IntegrateMeasurement`. This was mostly for two reasons: 1) `AddMeasurement` almost sounds like we're appending to a list and not really doing any calculation, whereas `IntegrateMeasurement` is more specific to this use case and I think conveys the idea that we're doing some math here (and it is a relatively expensive function after all). 2) `IntegrateMeasurement` is what gtsam calls it and, all things being equal, I'd rather have similar names to gtsam just to make our version easier to understand for someone who has experience with how gtsam uses it.

Reply:
After an in-person conversation, ended up just going with the constructor from a `Preintegrator<Scalar>`.